dsl/pace/dsl/gt4py_utils.py

```diff
@@ -355,9 +355,13 @@ def asarray(array, to_type=np.ndarray, dtype=None, order=None):
             return np.asarray(array, dtype, order)
         else:
             return cp.asarray(array, dtype, order)
-    if cp and (
-        isinstance(array, memoryview)
-        or isinstance(array.data, (cp.ndarray, cp.cuda.memory.MemoryPointer))
+    if (
+        cp
+        and hasattr(array, "data")
+        and (
+            isinstance(array, memoryview)
+            or isinstance(array.data, (cp.ndarray, cp.cuda.memory.MemoryPointer))
+        )
```

Comment:
The logic for this has changed, can you make sure it still evaluates to `True` if the object is a `memoryview`, even though memoryview objects have no "data" attribute?

You may be able to make use of `getattr(array, "data", None)` which will default to None if there is no attribute.

Reply:
fixed